.github/workflows/main.yml

```diff
@@ -53,29 +53,90 @@ jobs:
             just docker/build prod
             just docker/build dev
 
+      - name: Save docker prod image
+        run: |
+          docker save airlock | gzip > /tmp/airlock.tar.gz
+
+      - name: Upload docker prod image
+        uses: actions/upload-artifact@v4
+        with:
+            name: airlock-image
+            path: /tmp/airlock.tar.gz
+
+      - name: Save docker dev image
+        run: |
+          docker save airlock-dev | gzip > /tmp/airlock-dev.tar.gz
+
+      - name: Upload docker dev image
+        uses: actions/upload-artifact@v4
+        with:
+            name: airlock-dev-image
+            path: /tmp/airlock-dev.tar.gz
+
+
+
+  docker-dev-test-all:
+    needs: [docker-build]
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: "opensafely-core/setup-action@v1"
+        with:
+          install-just: true
+
+      - name: Download docker image
+        uses: actions/download-artifact@v4
+        with:
+            name: airlock-dev-image
+            path: /tmp/image
+
+      - name: Import docker image
+        run: gunzip -c /tmp/image/airlock-dev.tar.gz | docker load
+
       - name: Run unit tests on docker dev image
         run: |
           # build docker and run test
           just docker/test
 
+
+  docker-prod-smoke-test:
+    needs: [docker-build]
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: "opensafely-core/setup-action@v1"
+        with:
+          install-just: true
+ 
+      - name: Download docker image
+        uses: actions/download-artifact@v4
+        with:
+            name: airlock-image
+            path: /tmp/image
+
+      - name: Import docker image
+        run: gunzip -c /tmp/image/airlock.tar.gz | docker load
+
+      - name: create workdir
+        run: |
+            # the prod container is forcibly running as user 10000
+            # so make this world-readable for now so we can unblock next steps
+            sudo touch workdir/db.sqlite3
+            sudo chmod -R ugo+rw workdir/
```

Comment:
previously docker/smoke-test was passing in CI without this, but only because docker/test was touching these files with different permissions.
nb. I don't love this, open to alternative suggestions